src/ol_infrastructure/applications/airbyte/__main__.py

```diff
@@ -147,6 +157,118 @@
+        ),
+    ],
+    tags=aws_config.tags,
+    vpc_id=vpc_id,
+)
+
+#########################
+#     Database Setup    #
+#########################
+rds_defaults = defaults(stack_info)["rds"]
+rds_defaults["instance_size"] = (
+    airbyte_config.get("db_instance_size") or rds_defaults["instance_size"]
+)
+rds_password = read_yaml_secrets(
+    Path(f"data/airbyte/rds.{stack_info.env_suffix}.yaml")
+)["password"]
+
+airbyte_db_config = OLPostgresDBConfig(
+    instance_name=f"airbyte-db-{stack_info.env_suffix}",
+    password=rds_password,
+    storage=airbyte_config.get("db_capacity"),
+    subnet_group_name=target_vpc["rds_subnet"],
+    security_groups=[airbyte_db_security_group],
+    parameter_overrides=[{"name": "rds.force_ssl", "value": 0}],
+    tags=aws_config.tags,
+    db_name="airbyte",
+    engine_version="12.7",
+    **rds_defaults,
+)
+airbyte_db = OLAmazonDB(airbyte_db_config)
+
+# Shorten a few frequently used attributes from the database
+db_address = airbyte_db.db_instance.address
+db_port = airbyte_db.db_instance.port
+db_name = airbyte_db.db_instance.name
+
+airbyte_db_vault_backend_config = OLVaultPostgresDatabaseConfig(
+    db_name=airbyte_db_config.db_name,
+    mount_point=f"{airbyte_db_config.engine}-airbyte",
+    db_admin_username=airbyte_db_config.username,
+    db_admin_password=rds_password,
+    db_host=airbyte_db.db_instance.address,
+)
+airbyte_db_vault_backend = OLVaultDatabaseBackend(airbyte_db_vault_backend_config)
+
+airbyte_db_consul_node = Node(
+    "airbyte-instance-db-node",
+    name="airbyte-postgres-db",
+    address=db_address,
+    opts=consul_provider,
+)
+
+concourse_db_consul_service = Service(
+    "airbyte-instance-db-service",
+    node=airbyte_db_consul_node.name,
+    name="airbyte-postgres",
+    port=db_port,
+    meta={
+        "external-node": True,
+        "external-probe": True,
+    },
+    checks=[
+        ServiceCheckArgs(
+            check_id="airbyte-instance-db",
+            interval="10s",
+            name="airbyte-instance-db",
+            timeout="60s",
+            status="passing",
+            tcp=Output.all(
+                address=db_address,
+                port=db_port,
+            ).apply(lambda db: "{address}:{port}".format(**db)),
+        )
+    ],
+    opts=consul_provider,
+)
+
+connection_string = Output.all(address=db_address, port=db_port, name=db_name).apply(
+    lambda db: "jdbc:postgresql://{address}:{port}/{name}?ssl=true&sslmode=require".format(
+        **db
+    )
+)
+
+consul.Keys(
+    "airbyte-consul-template-data",
+    keys=[
+        consul.KeysKeyArgs(path=f"airbyte/database-host", value=db_address),
+        consul.KeysKeyArgs(path=f"airbyte/database-port", value=db_port),
+        consul.KeysKeyArgs(path=f"airbyte/database-name", value=db_name),
+        consul.KeysKeyArgs(
+            path=f"airbyte/database-connection-string", value=connection_string
+        ),
+    ],
+    opts=consul_provider,
+)
```

Comment:
Nothing to do for this PR, but this whole section seems to be a good candidate for wrapping in a function or component resource